### DISCUSSION TITLE
How to use ProviderIsAvailableOverride property in log providers

### DISCUSSION THREAD

**Discussion by MasterNewbie2** [quartznet/quartznet#954] [General] [answered]
We have an application that uses CSLA3, and we're trying to upgrade our Quartz to 3.1.0 so we can get the fix to https://github.com/quartznet/quartznet/issues/794.

We're getting a StackOverflowException in the Impl.DirectSchedulerFactory constructor using 3.1. The stack overflow is from mixing CSLA3's custom assembly resolve event handler with how the log provider is initialized. We were on 2.6.1, so we've been using NLog as our provider. The issue is that the log provider initialization tries to find the Serilog assembly first, and the custom assembly resolve handler is repeatedly recursively trying to load it.

Setting ProviderIsAvailableOverride to false for SerilogLogProvider would be sufficient to resolve this issue, but the class itself is internal. Is there some other way to set this?

**Comment by lahma** [comment, marked as answer]
I think you can work around this by calling `Quartz.Logging.LogProvider.SetCurrentLogProvider(ILogProvider)`. You you supply the `NLogLogProvider` (or your own wrapper) taken from sources it should skip trying to scan for the logger implementation as one is already set. If you are on .NET Core boat with MS logging abstractions and already have the NLog bridge there, there's also `Quartz.LogContext.SetCurrentLogProvider(ILoggerFactory)`.

**Reply by MasterNewbie2** [reply]
I reviewed the source and I agree that would probably work as a workaround.

We have a locally compiled version of the CSLA3 assembly so we can have side-by-side CSLA3 and CSLA5 so we don't have to migrate the entire app at one time to upgrade. We decided to change the custom assembly resolve handler to not result in a stack overflow when Quartz tries to load the other log provider assemblies.

Should I post this in issues? That the log provider overrides aren't publicly accessible.

**Reply by lahma** [reply]
I think they won't be opened at this point anymore. Quartz will migrate to MS logging abstraction as LibLog isn't continued anymore.
